diff --git a/arch/arm/boot/dts/qcom/apq8009-mtp-wcd9326-refboard.dts b/arch/arm/boot/dts/qcom/apq8009-mtp-wcd9326-refboard.dts
index a9c228a..3ba3053 100644
--- a/arch/arm/boot/dts/qcom/apq8009-mtp-wcd9326-refboard.dts
+++ b/arch/arm/boot/dts/qcom/apq8009-mtp-wcd9326-refboard.dts
@@ -93,3 +93,11 @@
 	status = "ok";
 	qcom,disable-bms;
 };
+
+&usb_otg {
+	interrupts = <0 134 0>,<0 140 0>,<0 136 0>;
+	interrupt-names = "core_irq", "async_irq", "phy_irq";
+
+	qcom,hsusb-otg-mode = <3>;
+	vbus_otg-supply = <&smb1360_otg_supply>;
+};
diff --git a/arch/arm/boot/dts/qcom/msm8909-mtp.dtsi b/arch/arm/boot/dts/qcom/msm8909-mtp.dtsi
index e809cac..0f1b0ea 100644
--- a/arch/arm/boot/dts/qcom/msm8909-mtp.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8909-mtp.dtsi
@@ -233,7 +233,7 @@
 };
 
 &blsp1_uart1 {
-	status = "disabled";
+	status = "ok";
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart_console_sleep>;
 };
diff --git a/drivers/media/usb/Makefile b/drivers/media/usb/Makefile
index 7f51d7e..857db7f 100644
--- a/drivers/media/usb/Makefile
+++ b/drivers/media/usb/Makefile
@@ -5,7 +5,7 @@
 # DVB USB-only drivers
 obj-y += ttusb-dec/ ttusb-budget/ dvb-usb/ dvb-usb-v2/ siano/ b2c2/
 obj-y += zr364xx/ stkwebcam/ s2255/
-
+CONFIG_USB_VIDEO_CLASS=y
 obj-$(CONFIG_USB_VIDEO_CLASS)	+= uvc/
 obj-$(CONFIG_USB_GSPCA)         += gspca/
 obj-$(CONFIG_USB_PWC)           += pwc/
diff --git a/drivers/media/usb/uvc/Makefile b/drivers/media/usb/uvc/Makefile
index c26d12f..7106305 100644
--- a/drivers/media/usb/uvc/Makefile
+++ b/drivers/media/usb/uvc/Makefile
@@ -1,3 +1,4 @@
+CONFIG_USB_VIDEO_CLASS=y
 uvcvideo-objs  := uvc_driver.o uvc_queue.o uvc_v4l2.o uvc_video.o uvc_ctrl.o \
 		  uvc_status.o uvc_isight.o uvc_debugfs.o
 ifeq ($(CONFIG_MEDIA_CONTROLLER),y)
